hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/avro/HoodieAvroParquetReader.java

```diff
@@ -50,24 +51,20 @@ public class HoodieAvroParquetReader extends RecordReader<Void, ArrayWritable> {
   private Schema baseSchema;
```

Comment:
In my [origin PR HUDI-83](https://github.com/apache/hudi/pull/3391/files) I didn't declare the `baseSchema` variable and didn't modify the `getCurrentValue` method.
In fact I would like to know if there is any problem if we don't declare the `baseSchema`?

Reply:
> In my [origin PR HUDI-83](https://github.com/apache/hudi/pull/3391/files) I didn't declare the `baseSchema` variable and didn't modify the `getCurrentValue` method. In fact I would like to know if there is any problem or no NPE if we don't declare the `baseSchema`?

I have tested that `baseSchema ` need to be used in getCurrentValue, otherwise, the result field will be null, like this https://github.com/apache/hudi/pull/7173#issuecomment-1316208189

Reply:
@Zouxxyy 
I'm having some confusion, I remember doing some situation testing against Hive when I first made the changes (about 1 year ago), including `count(*)`. 
I don't know if some subsequent new FEATURE or PR has affected this, I think I'll do another test later this week. Although we have added a separate class to handle `timestamp` types, my original intention was to use Hive or Hadoop origin method as much as possible for other fields, otherwise it would be costly for us to maintain subsequently.

Reply:
@cdmikechen 
Have you ever tested `select id, ts1 from test_ts_1`?  will return null if don't use `baseSchema`

Below is my full test, fell free to try

```sql
-- spark-sql
create table test_ts_1(
  id int, 
  ts1 timestamp)
using hudi
tblproperties(
  type='mor', 
  primaryKey='id'
);

INSERT INTO xinyu_test.test_ts_1
SELECT 1,
cast ('2021-12-25 12:01:01' as timestamp);

create table test_ts_2(
  id int, 
  ts1 array<timestamp>, 
  ts2 map<string, timestamp>, 
  ts3 struct<province:timestamp, city:string>)
using hudi
tblproperties(
  type='mor', 
  primaryKey='id'
);

INSERT INTO test_ts_2
SELECT 1,
array(cast ('2021-12-25 12:01:01' as timestamp)),
map('key', cast ('2021-12-25 12:01:01' as timestamp)),
struct(cast ('2021-12-25 12:01:01' as timestamp), 'test');

-- hive
select * from test_ts_1;
select id from test_ts_1;
select ts1 from test_ts_1;
select id, ts1 from test_ts_1;
select count(*) from test_ts_1;

select * from test_ts_2;
select id from test_ts_2;
select ts1 from test_ts_2;
select id, ts1 from test_ts_2;
select count(*) from test_ts_2;
```

CC @danny0405 @xicm